### ISSUE TITLE
Feature Request - allow special handling of diacritic in names

### ISSUE THREAD

**Issue by mkoo** [ArctosDB/arctos#7041]
This is from a previous issue where diacritics are needed for names in data entry. Currently requiring unicode to properly insert names for display on Project and other pages in Arctos

> <img alt="Screenshot 2023-12-08 at 9 12 20 AM" width="722" src="https://private-user-images.githubusercontent.com/17605945/289154231-fea26ecc-b47f-4478-9471-f27fbbc71e4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwNTk3MTYsIm5iZiI6MTcwMjA1OTQxNiwicGF0aCI6Ii8xNzYwNTk0NS8yODkxNTQyMzEtZmVhMjZlY2MtYjQ3Zi00NDc4LTk0NzEtZjI3ZmJiYzcxZTRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDE4MTY1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjMGUyOGVkZmNkMTcyY2IyODgwYzRlZTkzOGQ1NGU5YjJlYmU3OWU1NTYwZjhiZmFkZjlkOWNlODAwZTJhMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g7WNCJag3ikCvP-UqFgkQ-7fhizopEpxldCLfVlAwzU">
> 
> It's just an apostrophe... what happens when we get names of tribes that have other diacritic marks in their names?

@dustymc : what's the best solution to include diacritics in names (not just agent names)? This is needed in a lot of geographic and names databases so we dont have to reinvent the wheel... I'm just brainstorming here so adjust your expectations! We could include a diacritic translator help window and/or adopt a unicode converter in data entry? This also implies that we need to store some accent-free version of names so that searching on a name with or without diacritics will appear. So maybe a display name field needs to be added? We could make the 5 or so most common diacritics as special cases so they can be optioned in as needed (apostrophes, accent grave, accent acute, umlauts, or what have you)? This will probably introduce issues with exporting to CSV since some will be with the html-ready formatting and some versions will lack the diacritic. FYI most geographic databases have two fields for names-- one without diacritics and and one with.

Definitely filing under needing more technical discussion

Also a consideration if we create exports to bibliographic aggregators (eventually)
(OCLC https://help.oclc.org/Metadata_Services/WorldShare_Record_Manager/Bibliographic_records/MARC_21_view/Insert_a_diacritic)

**Comment by dustymc** [comment]
I believe this is only an issue on internal picks, where several languages (and dialects) are involved, and then only pre-save. I can probably do something else there if someone has specific requests, but I also expect that any non-native handling will be more fragile than what I'm doing now.

**Comment by AJLinn** [comment]
> Also, If you don't supply a de-diacritic-ed name in Agents, you get a warning that tells you to add one.

I also find this problematic - even when we supply a seemingly non-diactric version, the Agent error messenger continues to post errors in the notifications (this one has been showing up for months now):

<img width="1618" alt="Screenshot 2023-12-12 at 8 25 11 AM" src="https://github.com/ArctosDB/arctos/assets/17605945/cbe0704b-971d-4b98-881d-dfd724e260b5">

**Comment by Jegelewicz** [comment]
I suspect that is because I failed to add a non-ASCII variant for the middle name. I've done that now, let's see if it works...

![image](https://github.com/ArctosDB/arctos/assets/5725767/abe9296a-cbef-4432-8267-77642213a5da)

**Comment by dustymc** [comment]
Arctos is UTF8 encoded: https://handbook.arctosdb.org/documentation/encoding.html
